### PR TITLE
feat(clerk-js): Render `@clerk/ui`

### PR DESCRIPTION
**Pull request by BRKalow**
## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/nextjs/src/app-router/client/ClerkProvider.tsx
Comment (nikosdouvlis) [inline packages/nextjs/src/app-router/client/ClerkProvider.tsx:53]: Can you please provide some more context about this change?

Comment (panteliselef) [reply packages/nextjs/src/app-router/client/ClerkProvider.tsx:53]: I had the same question

Comment (BRKalow) [reply packages/nextjs/src/app-router/client/ClerkProvider.tsx:53]: @nikosdouvlis @panteliselef sure! With the way we are passing the router into clerk-js, these values become stale as we are closing over the values at the time of initial render in the provider here. The only way to get these values from Next is through the hooks, so we would need to update this router object on every render and ensure the clerk components re-render based on that.

As our components only render on the client, the source of truth _should_ be the URL, so it seems safe to do that here. This way it ensures the value is always fresh.

Comment (colinclerk) [reply packages/nextjs/src/app-router/client/ClerkProvider.tsx:53]: We've historically had problems with trying to use the `window.location` instead of what the router believes is currently the location.

The challenge is that each router implements its own method of changing pages. It's an async process that, at a minimum, requires `fetch()`ing the next page and rerendering.

The exact timing of when `window.history.pushState` gets called is an implementation detail, so looking at changes there isn't a tremendously reliable signal for the page having changed.  The inconsistency in implementations led to us creating an `awaitableNavigate()` in various SDKs.

All this is to say - I agree the source of _should_ be the URL, and to some extent it is, but the lack of precision means we should at least be very cautious here. Was there anything motivating this change from a functionality perspective, or was it changed optimistically as a cleaner approach?

Comment (nikosdouvlis) [reply packages/nextjs/src/app-router/client/ClerkProvider.tsx:53]: > but the lack of precision means we should at least be very cautious here

For the reasons Colin stated above, I'd prefer we test the change in isolation on a separate PR. I'd like to merge this today to unblock us from running the e2e using the new components and then spend some time verifying this works.

Comment (BRKalow) [reply packages/nextjs/src/app-router/client/ClerkProvider.tsx:53]: Guys, this change only impacts the router used by elements and the rebuilt components, can we agree to merge and do further testing?

File: packages/clerk-js/webpack.config.js
Comment (dstaley) [inline packages/clerk-js/webpack.config.js:274]: This was added due to the inclusion of Webpack's [auto public path runtime module](https://github.com/webpack/webpack/blob/dae16ad11e2d2accddc549a92082da3d2e667b01/lib/runtime/AutoPublicPathRuntimeModule.js#L74), which was unable to calculate a public path in our Chrome extension tests. Since this only impacts the CommonJS build, switching the chunk format to `commonjs` causes Webpack to emit chunks in CJS rather than in a format requiring a URL (I believe it defaulted to `importScripts` which is used for Web Workers).

**Comment by nikosdouvlis** [comment]
Hey folks! Shall we merge this one in today and continue testing? 
I'd like to enable the e2e tests to run against the AIO components as well within the day if possible :)

File: packages/clerk-js/src/ui/new/renderer.tsx
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/new/renderer.tsx:64]: Just wanted to point out here that after this refactor we're probably losing the ability to `await` the mount call.

This is something not blocking this PR at the moment, but I'd like to verify if that's the case and see if we actually need to ever `await` it or not

**Review by nikosdouvlis — APPROVED** [review]

**Comment by nikosdouvlis** [comment]
Merging so we can start testing

File: packages/clerk-js/package.json
Comment (jacekradko) [inline packages/clerk-js/package.json:55]: [nit] Non-module packages, like types, are dev deps so they are not needed to be installed by consumers. Should be safe to move this to `devDeps`

File: packages/clerk-js/src/ui/new/index.tsx
Comment (jacekradko) [inline packages/clerk-js/src/ui/new/index.tsx:8]: I assume this it temporary to get past some typing issues?

File: packages/ui/src/components/sign-up/sign-up.tsx
Comment (jacekradko) [inline packages/ui/src/components/sign-up/sign-up.tsx:9]: [minor] I think these are equivalent and don't require 2 lines

```suggestion
import { type Appearance, AppearanceProvider } from '~/contexts';
```

File: packages/clerk-js/package.json
Comment (jacekradko) [inline packages/clerk-js/package.json:53]: Looks like a git conflict issue, as this line is duplicated on L:56

```suggestion
```

File: packages/clerk-js/package.json
Comment (jacekradko) [inline packages/clerk-js/package.json:54]: Same here, git conflict resolution issues

```suggestion
```

File: packages/clerk-js/package.json
Comment (jacekradko) [reply packages/clerk-js/package.json:55]: I was just looking at `@clerk/types` and it isn't actually a types-only package as it exports a JS module. That's a bit confusing. @BRKalow thoughts?

**Comment by dstaley** [comment]
!snapshot

**Comment by dstaley** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @dstaley - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/astro | 1.3.13-snapshot.vd415f19 |
| @clerk/backend | 1.13.9-snapshot.vd415f19 |
| @clerk/chrome-extension | 1.3.15-snapshot.vd415f19 |
| @clerk/clerk-js | 5.27.0-snapshot.vd415f19 |
| @clerk/elements | 0.16.0-snapshot.vd415f19 |
| @clerk/clerk-expo | 2.2.21-snapshot.vd415f19 |
| @clerk/express | 1.2.2-snapshot.vd415f19 |
| @clerk/fastify | 1.0.52-snapshot.vd415f19 |
| @clerk/localizations | 3.1.3-snapshot.vd415f19 |
| @clerk/nextjs | 5.8.0-snapshot.vd415f19 |
| @clerk/clerk-react | 5.11.1-snapshot.vd415f19 |
| @clerk/remix | 4.2.36-snapshot.vd415f19 |
| @clerk/clerk-sdk-node | 5.0.49-snapshot.vd415f19 |
| @clerk/shared | 2.10.0-snapshot.vd415f19 |
| @clerk/tanstack-start | 0.4.12-snapshot.vd415f19 |
| @clerk/testing | 1.3.10-snapshot.vd415f19 |
| @clerk/themes | 2.1.36-snapshot.vd415f19 |
| @clerk/types | 4.26.0-snapshot.vd415f19 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/astro`
```sh
npm i @clerk/astro@1.3.13-snapshot.vd415f19 --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@1.13.9-snapshot.vd415f19 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.3.15-snapshot.vd415f19 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.27.0-snapshot.vd415f19 --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.16.0-snapshot.vd415f19 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@2.2.21-snapshot.vd415f19 --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@1.2.2-snapshot.vd415f19 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@1.0.52-snapshot.vd415f19 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@3.1.3-snapshot.vd415f19 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.8.0-snapshot.vd415f19 --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.11.1-snapshot.vd415f19 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.2.36-snapshot.vd415f19 --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.49-snapshot.vd415f19 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.10.0-snapshot.vd415f19 --save-exact
```
`@clerk/tanstack-start`
```sh
npm i @clerk/tanstack-start@0.4.12-snapshot.vd415f19 --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.3.10-snapshot.vd415f19 --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.1.36-snapshot.vd415f19 --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.26.0-snapshot.vd415f19 --save-exact
```

**Comment by dstaley** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @dstaley - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/astro | 1.3.13-snapshot.v477ee05 |
| @clerk/backend | 1.13.9-snapshot.v477ee05 |
| @clerk/chrome-extension | 1.3.15-snapshot.v477ee05 |
| @clerk/clerk-js | 5.27.0-snapshot.v477ee05 |
| @clerk/elements | 0.16.0-snapshot.v477ee05 |
| @clerk/clerk-expo | 2.2.21-snapshot.v477ee05 |
| @clerk/express | 1.2.2-snapshot.v477ee05 |
| @clerk/fastify | 1.0.52-snapshot.v477ee05 |
| @clerk/localizations | 3.1.3-snapshot.v477ee05 |
| @clerk/nextjs | 5.8.0-snapshot.v477ee05 |
| @clerk/clerk-react | 5.11.1-snapshot.v477ee05 |
| @clerk/remix | 4.2.36-snapshot.v477ee05 |
| @clerk/clerk-sdk-node | 5.0.49-snapshot.v477ee05 |
| @clerk/shared | 2.10.0-snapshot.v477ee05 |
| @clerk/tanstack-start | 0.4.12-snapshot.v477ee05 |
| @clerk/testing | 1.3.10-snapshot.v477ee05 |
| @clerk/themes | 2.1.36-snapshot.v477ee05 |
| @clerk/types | 4.26.0-snapshot.v477ee05 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/astro`
```sh
npm i @clerk/astro@1.3.13-snapshot.v477ee05 --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@1.13.9-snapshot.v477ee05 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.3.15-snapshot.v477ee05 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.27.0-snapshot.v477ee05 --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.16.0-snapshot.v477ee05 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@2.2.21-snapshot.v477ee05 --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@1.2.2-snapshot.v477ee05 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@1.0.52-snapshot.v477ee05 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@3.1.3-snapshot.v477ee05 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.8.0-snapshot.v477ee05 --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.11.1-snapshot.v477ee05 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.2.36-snapshot.v477ee05 --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.49-snapshot.v477ee05 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.10.0-snapshot.v477ee05 --save-exact
```
`@clerk/tanstack-start`
```sh
npm i @clerk/tanstack-start@0.4.12-snapshot.v477ee05 --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.3.10-snapshot.v477ee05 --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.1.36-snapshot.v477ee05 --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.26.0-snapshot.v477ee05 --save-exact
```

File: packages/clerk-js/bundlewatch.config.json
Comment (panteliselef) [inline packages/clerk-js/bundlewatch.config.json:18]: ❓ how's this affected

File: packages/clerk-js/package.json
Comment (jacekradko) [inline packages/clerk-js/package.json:56]: [question] is `@clerk/ui` considered pre-release since it's still `0.x.x`?

Comment (dstaley) [reply packages/clerk-js/package.json:56]: Yup, it's an internal-only package essentially. At this point I'm not aware of any plans to make it a publicly consumable package, so we might be able to get away with keeping it private and pinning the version to `"*"`. I'll let @BRKalow confirm though.

The `ui` package is bundled via Webpack into `clerk-js`, so we don't need it published to npm in order to consume it.

**Comment by dstaley** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @dstaley - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/astro | 1.3.13-snapshot.v860897c |
| @clerk/backend | 1.13.9-snapshot.v860897c |
| @clerk/chrome-extension | 1.3.15-snapshot.v860897c |
| @clerk/clerk-js | 5.27.0-snapshot.v860897c |
| @clerk/elements | 0.16.0-snapshot.v860897c |
| @clerk/clerk-expo | 2.2.21-snapshot.v860897c |
| @clerk/express | 1.2.2-snapshot.v860897c |
| @clerk/fastify | 1.0.52-snapshot.v860897c |
| @clerk/localizations | 3.1.3-snapshot.v860897c |
| @clerk/nextjs | 5.8.0-snapshot.v860897c |
| @clerk/clerk-react | 5.11.1-snapshot.v860897c |
| @clerk/remix | 4.2.36-snapshot.v860897c |
| @clerk/clerk-sdk-node | 5.0.49-snapshot.v860897c |
| @clerk/shared | 2.10.0-snapshot.v860897c |
| @clerk/tanstack-start | 0.4.12-snapshot.v860897c |
| @clerk/testing | 1.3.10-snapshot.v860897c |
| @clerk/themes | 2.1.36-snapshot.v860897c |
| @clerk/types | 4.26.0-snapshot.v860897c |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/astro`
```sh
npm i @clerk/astro@1.3.13-snapshot.v860897c --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@1.13.9-snapshot.v860897c --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.3.15-snapshot.v860897c --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.27.0-snapshot.v860897c --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.16.0-snapshot.v860897c --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@2.2.21-snapshot.v860897c --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@1.2.2-snapshot.v860897c --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@1.0.52-snapshot.v860897c --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@3.1.3-snapshot.v860897c --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.8.0-snapshot.v860897c --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.11.1-snapshot.v860897c --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.2.36-snapshot.v860897c --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.49-snapshot.v860897c --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.10.0-snapshot.v860897c --save-exact
```
`@clerk/tanstack-start`
```sh
npm i @clerk/tanstack-start@0.4.12-snapshot.v860897c --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.3.10-snapshot.v860897c --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.1.36-snapshot.v860897c --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.26.0-snapshot.v860897c --save-exact
```

**Comment by dstaley** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @dstaley - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/astro | 1.3.13-snapshot.v6bae6c3 |
| @clerk/backend | 1.13.9-snapshot.v6bae6c3 |
| @clerk/chrome-extension | 1.3.16-snapshot.v6bae6c3 |
| @clerk/clerk-js | 5.27.0-snapshot.v6bae6c3 |
| @clerk/elements | 0.16.0-snapshot.v6bae6c3 |
| @clerk/clerk-expo | 2.2.22-snapshot.v6bae6c3 |
| @clerk/express | 1.2.2-snapshot.v6bae6c3 |
| @clerk/fastify | 1.0.52-snapshot.v6bae6c3 |
| @clerk/localizations | 3.2.0-snapshot.v6bae6c3 |
| @clerk/nextjs | 5.8.0-snapshot.v6bae6c3 |
| @clerk/clerk-react | 5.11.1-snapshot.v6bae6c3 |
| @clerk/remix | 4.2.36-snapshot.v6bae6c3 |
| @clerk/clerk-sdk-node | 5.0.49-snapshot.v6bae6c3 |
| @clerk/shared | 2.10.0-snapshot.v6bae6c3 |
| @clerk/tanstack-start | 0.4.12-snapshot.v6bae6c3 |
| @clerk/testing | 1.3.10-snapshot.v6bae6c3 |
| @clerk/themes | 2.1.36-snapshot.v6bae6c3 |
| @clerk/types | 4.26.0-snapshot.v6bae6c3 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/astro`
```sh
npm i @clerk/astro@1.3.13-snapshot.v6bae6c3 --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@1.13.9-snapshot.v6bae6c3 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.3.16-snapshot.v6bae6c3 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.27.0-snapshot.v6bae6c3 --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.16.0-snapshot.v6bae6c3 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@2.2.22-snapshot.v6bae6c3 --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@1.2.2-snapshot.v6bae6c3 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@1.0.52-snapshot.v6bae6c3 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@3.2.0-snapshot.v6bae6c3 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.8.0-snapshot.v6bae6c3 --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.11.1-snapshot.v6bae6c3 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.2.36-snapshot.v6bae6c3 --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.49-snapshot.v6bae6c3 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.10.0-snapshot.v6bae6c3 --save-exact
```
`@clerk/tanstack-start`
```sh
npm i @clerk/tanstack-start@0.4.12-snapshot.v6bae6c3 --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.3.10-snapshot.v6bae6c3 --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.1.36-snapshot.v6bae6c3 --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.26.0-snapshot.v6bae6c3 --save-exact
```

**Comment by dstaley** [comment]
!snapshot

**Comment by dstaley** [comment]
!snapshot

**Pull request merged**
